identity_stardust/src/did/stardust_did.rs

```diff
@@ -0,0 +1,929 @@
+  //
+  // Also consider replacing this with a `setter` in order to be more similar to the JS API.
+  pub fn with_network(_network: &str) -> Self {
+    todo!("implement this once the network/network name has been ported")
+  }
+}
+
+impl DID for StardustDID {
+  /// Returns the [`StardustDID`] scheme. See [`DID::SCHEME`].
+  fn scheme(&self) -> &'static str {
+    self.0.scheme()
+  }
+
+  /// Returns the [`StardustDID`] authority.
+  fn authority(&self) -> &str {
+    self.0.authority()
+  }
+
+  /// Returns the [`StardustDID`] method name.
+  fn method(&self) -> &str {
+    self.0.method()
+  }
+
+  /// Returns the [`StardustDID`] method-specific ID.
+  fn method_id(&self) -> &str {
+    self.0.method_id()
+  }
+
+  /// Returns the serialized [`StardustDID`].
+  ///
+  /// This is fast since the serialized value is stored in the [`DID`].
+  fn as_str(&self) -> &str {
+    self.0.as_str()
+  }
+
+  /// Consumes the [`StardustDID`] and returns the serialization.
+  fn into_string(self) -> String {
+    self.0.into_string()
+  }
+
+  /// Creates a new [`StardustDIDUrl`] by joining with a relative DID Url string.
+  ///
+  /// # Errors
+  ///
+  /// Returns `Err` if any base or relative DID segments are invalid.
+  fn join(self, segment: impl AsRef<str>) -> std::result::Result<DIDUrl<Self>, DIDError> {
+    self.into_url().join(segment)
+  }
+
+  fn to_url(&self) -> DIDUrl<Self> {
+    DIDUrl::new(self.clone(), None)
+  }
+
+  fn into_url(self) -> DIDUrl<Self> {
+    DIDUrl::new(self, None)
+  }
+}
+
+impl FromStr for StardustDID {
+  type Err = DIDError;
+
+  fn from_str(s: &str) -> std::result::Result<Self, Self::Err> {
+    Self::parse(s)
+  }
+}
+
+impl TryFrom<&str> for StardustDID {
+  type Error = DIDError;
+
+  fn try_from(other: &str) -> std::result::Result<Self, Self::Error> {
+    Self::parse(other)
+  }
+}
+
+impl TryFrom<String> for StardustDID {
+  type Error = DIDError;
+
+  fn try_from(other: String) -> std::result::Result<Self, Self::Error> {
+    Self::parse(other)
+  }
+}
+
+impl Display for StardustDID {
+  fn fmt(&self, f: &mut Formatter<'_>) -> core::fmt::Result {
+    write!(f, "{}", self.0)
+  }
+}
+
+impl From<StardustDID> for CoreDID {
+  fn from(id: StardustDID) -> Self {
+    id.0
+  }
+}
+
+impl From<StardustDID> for String {
+  fn from(did: StardustDID) -> Self {
+    did.into_string()
+  }
+}
+
+impl From<AliasId> for StardustDID {
+  /// Transforms an [`AliasId`] to a [`StardustDID`].
+  ///
+  /// # Network
+  /// The [`StardustDID`] constructed from this method is assumed to be associated with the default network,
+  /// whenever that is not the case one should follow this up with calling [`StardustDID::with_network`].  
+  fn from(id: AliasId) -> Self {
+    let did_str = format!("did:{}:{}", StardustDID::METHOD, id);
+    Self::parse(did_str).unwrap_or_else(|_| {
+      panic!(
+        "transforming an AliasId to a {} DID should be infallible",
+        StardustDID::METHOD
+      )
+    })
+  }
+}
```

Comment:
We may not want this due to the `network` requirement. I think something like a feature-gated `from_alias_id(alias_id, Option<network>)` function might be better?

Reply:
I will create a new issue about `Network` related tasks that need to be addressed in `StardustDID` once the network related functionality has been ported.

Reply:
Edit: Decided to introduce `NetworkName` now in this PR to deal with these things right away.